ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Mempool/Query.hs

```diff
@@ -26,30 +30,36 @@ implGetSnapshotFor ::
      , HasTxId (GenTx blk)
      )
   => MempoolEnv m blk
-  -> Point blk -- ^ The point on the ledger database where we want to acquire a
-               -- snapshot on
   -> SlotNo -- ^ Get snapshot for this slot number (usually the current slot)
   -> TickedLedgerState blk DiffMK -- ^ The ledger state at 'pt' ticked to 'slot'
-  -> m (Maybe (MempoolSnapshot blk))
-implGetSnapshotFor mpEnv pt slot ticked = do
+  -> LedgerTables (LedgerState blk) SeqDiffMK
+  -> LedgerBackingStoreValueHandle m (LedgerState blk)
+  -> m (MempoolSnapshot blk)
+implGetSnapshotFor mpEnv slot ticked chlog (LedgerBackingStoreValueHandle vhSlot vh) = do
   is <- atomically $ readTMVar istate
   if pointHash (isTip is) == castHash (getTipHash ticked) &&
      isSlotNo is == slot
     then
       -- We are looking for a snapshot exactly for the ledger state we already
       -- have cached, then just return it.
-      pure . Just . snapshotFromIS $ is
+      pure . snapshotFromIS $ is
     else do
-      -- We need to revalidate the transactions.
-      let txs = [ txForgetValidated . TxSeq.txTicketTx $ tx
-                | tx <- TxSeq.toList $ isTxs is
-                ]
-          go = do
-            mTbs <- getLedgerTablesAtFor ldgrInterface pt txs
-            case mTbs of
-              Right tbs            -> pure $ Just $ getSnap is tbs
-              Left PointNotFound{} -> pure Nothing
-      go
+       let keys = foldl' (zipLedgerTables (<>)) emptyLedgerTables
```

Comment:
You could use the `Monoid` instance for ledger tables from https://github.com/input-output-hk/ouroboros-consensus/blob/57489d3204139b5f57c9be72854a063bac1e7d3e/ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Ledger/Tables.hs#L348